packages/lesswrong/server/manualMigrations/2025-02-26-fieldChangesPopulateTable.ts

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable no-console */
+import { registerMigration } from "./migrationUtils";
+import LWEvents from "@/lib/collections/lwevents/collection";
+import { executePromiseQueue, sleep } from "@/lib/utils/asyncUtils";
+import { randomId } from "@/lib/random";
+import { FieldChanges } from "@/lib/collections/fieldChanges/collection";
+
+export default registerMigration({
+  name: "fieldChangesPopulateTable",
+  dateWritten: "2025-02-26",
+  idempotent: true,
+  action: async () => {
+    // Transfers field-changes from LWEvents to the dedicated FieldChanges
+    // collection. Note that LWEvents is an _extremely_ large table, mostly due
+    // to event types that are not this one, and queries against it must be
+    // made with great care lest they time out and cause problems for the
+    // database server.
+    // Also note that development DBs may be created by cloning a production
+    // DB with LWEvents omitted, which complicates performance testing. 
+
+    console.log("Loading field changes from the LWEvents table");
+    const fieldChangesFromLWEvents = await LWEvents.find({
+      name: "fieldChanges"
+    }).fetch();
+    
+    console.log(`Got ${fieldChangesFromLWEvents.length} field-change events`);
+    console.log("Inserting field changes into the FieldChanges table");
+    await executePromiseQueue(fieldChangesFromLWEvents.map((lwEvent) => async () => {
+      const changeGroupId = randomId();
```

Comment:
Possibly, for the migration, this should be some kind of deterministic hashing of the original record (i.e. of the documentId + createdAt), so that the migration is idempotent and can be resumed if it fails partway through for some reason.

Reply:
🤦 actually this can surely just be the original `_id` of the LWEvent, no?